### ISSUE TITLE
Adding querystring validator

### ISSUE THREAD

**Issue by jenstroeger** [Cornices/cornice#430]
Cornice provides a [body validator](https://github.com/Cornices/cornice/blob/master/cornice/validators/_colander.py#L9) which validates the body of a request against a [Colander](https://github.com/Pylons/colander) schema.

In order to validate the query string of a request against a schema, however, it takes [an extra indirection](http://cornice.readthedocs.io/en/latest/schema.html#multiple-request-attributes) through a Colander mapping schema, and the returned `request.validated` dictionary contains that same indirection.

Wouldn't it be easier (and make much sense) to add another dedicated query string validator to Cornice? And perhaps do the same for headers?

For example, suppose I have a view function to handle a `GET` which takes only a query string argument. A dedicated query string validator would be more explicit and convenient:

```python
@my_service.get(
    schema=SomeSchema(),
    validators=colander_querystring_validator,
    )
def get_all_foos(request):                                               
    ...
    foo = request.validated["foo"] # instead of request.validated["querystring"]["foo"]
    ...
```

Only if request data from different parts of the request needs validation would I need the generic `colander_validator`.

**Comment by Natim** [comment]
Hi @jenstroeger would you like to give it a try and start a PR on this one?

**Comment by jenstroeger** [comment]
@Natim, I'd like to but won't be able to dedicate time to it in the next week or two...

**Comment by Natim** [comment]
I guess we can wait for a couple of weeks :+1:

**Comment by jenstroeger** [comment]
Well, I haven’t had any time for this and likely won’t in the near future. Any chance somebody can pick this up and introduce consistency to the different Colander validators?

**Comment by jenstroeger** [comment]
It looks to me as if issue https://github.com/Cornices/cornice/issues/449#issuecomment-315312966 is related to this one, and would motivate different explicit Colander validators.

**Comment by jenstroeger** [comment]
> @leplatrem https://github.com/Cornices/cornice/issues/430 is a good way to start, rather small and scoped :)

Agreed, and I’m already working on it. I think for this change it would also make sense to _discontinue_ support of `querystring` in the schema. Certainly will simplify my code quite a bit...

**Comment by leplatrem** [comment]
> Agreed, and I’m already working on it. I

Great!

> it would also make sense to discontinue support of querystring in the schema

Oh, really?

> Certainly will simplify my code quite a bit...

Do hesitate to submit a PR with your changes so that we can see what we can do!

**Comment by jenstroeger** [comment]
Looks like pull request https://github.com/Cornices/cornice/pull/465 takes care of this one 🤔

**Comment by leplatrem** [comment]
Fixed by #465 indeed
